### ISSUE TITLE
Application can't find runtime when installing/reinstalling it for the first time

### ISSUE THREAD

**Issue by garuma** [xamarin/xamarin-android#1395]
### Steps to Reproduce

1. Phone with no Mono shared runtime installed
2. Compile and run Moyeu

Monodroid master 918e0b8e6d4c51c3ef563152e6ba516bc2b5c9a8

Output:

```
Forwarding debugger port 8816
Detecting existing process
> am start -n "org.neteril.Moyeu/md5bed1eba2499d8c032005c73fd288d4f3.MainActivity"
> Starting: Intent { cmp=org.neteril.Moyeu/md5bed1eba2499d8c032005c73fd288d4f3.MainActivity }

[zygote64] Late-enabling -Xcheck:jni
[monodroid] Creating public update directory: `/data/user/0/org.neteril.Moyeu/files/.__override__`
[monodroid] Using override path: /data/user/0/org.neteril.Moyeu/files/.__override__
[monodroid] Using override path: /storage/emulated/0/Android/data/org.neteril.Moyeu/files/.__override__
[monodroid] Trying to load sgen from: /data/user/0/org.neteril.Moyeu/files/.__override__/libmonosgen-2.0.so
[monodroid] Trying to load sgen from: /storage/emulated/0/Android/data/org.neteril.Moyeu/files/.__override__/libmonosgen-2.0.so
[monodroid] Trying to load sgen from: /storage/emulated/0/../legacy/Android/data/org.neteril.Moyeu/files/.__override__/libmonosgen-2.0.so
[monodroid] Trying to load sgen from: /data/app/org.neteril.Moyeu-9z-3mJdiFWt4yHyh5BwKLQ==/lib/arm64/libmonosgen-2.0.so
[monodroid] Trying to load sgen from: /data/user/0/org.neteril.Moyeu/files/.__override__/links/libmonosgen-2.0.so
[monodroid] Trying to load sgen from: /system/lib/libmonosgen-2.0.so
[monodroid] cannot find libmonosgen-2.0.so in override_dir: /data/user/0/org.neteril.Moyeu/files/.__override__, app_libdir: /data/app/org.neteril.Moyeu-9z-3mJdiFWt4yHyh5BwKLQ==/lib/arm64 nor in previously printed locations.
[monodroid] Do you have a shared runtime build of your app with AndroidManifest.xml android:minSdkVersion < 10 while running on a 64-bit Android 5.0 target? This combination is not supported.
[monodroid] Please either set android:minSdkVersion >= 10 or use a build without the shared runtime (like default Release configuration).
```

Deploy log:

```
Build started 3/11/2018 2:34:41 PM.
__________________________________________________
Project "/Users/jeremie/projects/Moyeu/Moyeu/Moyeu.csproj" (Install target(s)):

Target _SetLatestTargetFrameworkVersion:
    Found Java SDK version 1.8.0.
Target _ValidateAndroidPackageProperties:
      PackageName: org.neteril.Moyeu
Target _ResolveMonoAndroidSdks:
    MonoAndroid Tools: /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/
    Android Platform API level: 27
    TargetFrameworkVersion: v8.1
    Android NDK: /
    Android SDK: /Users/jeremie/android-sdk-macosx/
    Android SDK Build Tools: /Users/jeremie/android-sdk-macosx/build-tools/27.0.3/
    Java SDK: /usr/
Target _GetPrimaryCpuAbi:
    _PrimaryCpuAbi=
Target _CheckInstantRunCondition:
    Dex Fast Deployment Enabled: False
Target _SetupInstantRun:
    Android Java Class for Fast Deployment: 
Target _SetupApplicationJavaClass:
    Application Java class: android.app.Application
Target ProcessGoogleServicesJsonChecks:
    No GoogleServicesJson file specified.
Target _GetProjectReferenceTargetFrameworkProperties:
    __________________________________________________
    Project "/Users/jeremie/projects/Moyeu/Moyeu/Moyeu.csproj" is building "/Users/jeremie/projects/Moyeu/HtmlParserSharp/HtmlParserSharp.csproj" (GetTargetFrameworks target(s)):

Target ResolveProjectReferences:
    __________________________________________________
    Project "/Users/jeremie/projects/Moyeu/Moyeu/Moyeu.csproj" is building "/Users/jeremie/projects/Moyeu/HtmlParserSharp/HtmlParserSharp.csproj" (GetTargetPath target(s)):

    __________________________________________________
    Project "/Users/jeremie/projects/Moyeu/Moyeu/Moyeu.csproj" is building "/Users/jeremie/projects/Moyeu/HtmlParserSharp/HtmlParserSharp.csproj" (GetNativeManifest target(s)):

Target _XamarinAndroidBuildAarRestore:
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Basement.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Tasks.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Base.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Wearable.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Maps.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Places.dll.stamp
    Reference has already had resources merged, skipping due to: obj/Debug/XbdMerge/Xamarin.GooglePlayServices.Location.dll.stamp
Target _ResolveLibraryProjectImports:
  Skipping target "_ResolveLibraryProjectImports" because all output files are up-to-date with respect to the input files.
Target _BuildLibraryImportsCache:
  Skipping target "_BuildLibraryImportsCache" because all output files are up-to-date with respect to the input files.
Target _BuildAdditionalResourcesCache:
  Skipping target "_BuildAdditionalResourcesCache" because all output files are up-to-date with respect to the input files.
Target _CreateAdditionalResourceCache:
  Skipping target "_CreateAdditionalResourceCache" because it has no outputs.
Target _GenerateAndroidResourceDir:
  Skipping target "_GenerateAndroidResourceDir" because all output files are up-to-date with respect to the input files.
Target _GenerateCodeBehindForLayout:
  Skipping target "_GenerateCodeBehindForLayout" because it has no outputs.
Target _DefineBuildTargetAbis:
    Build target ABI: arm64-v8a;armeabi;armeabi-v7a;x86;x86_64
Target VectorDrawableCheckBuildToolsVersionTask:
    Checking Android SDK Build-tools version...
    Selected Android SDK Build Tools Path: /Users/jeremie/android-sdk-macosx/build-tools/27.0.3/
    Selected Android SDK Build Tools Version: 27.0.3
    Android SDK Build Tools Version: 27.0.3 meets minimum requirements for Vector Drawables. OK.
    Finished Checking Android SDK Build-tools version.
Target _UpdateAndroidResgen:
  Skipping target "_UpdateAndroidResgen" because all output files are up-to-date with respect to the input files.
Target _CreateAdditionalResourceCache:
  Skipping target "_CreateAdditionalResourceCache" because it has no outputs.
Target _GenerateJavaDesignerForComponent:
  Skipping target "_GenerateJavaDesignerForComponent" because all output files are up-to-date with respect to the input files.
Target GenerateTargetFrameworkMonikerAttribute:
  Skipping target "GenerateTargetFrameworkMonikerAttribute" because all output files are up-to-date with respect to the input files.
Target CoreCompile:
  Skipping target "CoreCompile" because all output files are up-to-date with respect to the input files.
Target _CopyFilesMarkedCopyLocal:
    Touching "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/Moyeu.csproj.CopyComplete".
Target GetCopyToOutputDirectoryItems:
    __________________________________________________
    Project "/Users/jeremie/projects/Moyeu/Moyeu/Moyeu.csproj" is building "/Users/jeremie/projects/Moyeu/HtmlParserSharp/HtmlParserSharp.csproj" (GetCopyToOutputDirectoryItems target(s)):

Target CopyFilesToOutputDirectory:
    Moyeu -> /Users/jeremie/projects/Moyeu/Moyeu/bin/Debug/Moyeu.dll
Target IncrementalClean:
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/linksrc/Moyeu.pdb".
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/linksrc/ActivityTask.pdb".
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/linksrc/HtmlParserSharp.pdb".
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/linksrc/Mono.Android.pdb".
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/bin/Debug/ActivityTask.pdb".
    Deleting file "/Users/jeremie/projects/Moyeu/Moyeu/bin/Debug/Mono.Android.pdb".
Target _CollectConfigFiles:
  Skipping target "_CollectConfigFiles" because all output files are up-to-date with respect to the input files.
Target _CopyConfigFiles:
  Skipping target "_CopyConfigFiles" because it has no outputs.
Target _CopyIntermediateAssemblies:
  Skipping target "_CopyIntermediateAssemblies" because all output files are up-to-date with respect to the input files.
Target _ConvertPdbFiles:
  Skipping target "_ConvertPdbFiles" because it has no outputs.
Target _CopyPdbFiles:
    Copying file from "bin/Debug/Moyeu.pdb" to "obj/Debug/linksrc/Moyeu.pdb".
    Copying file from "bin/Debug/ActivityTask.pdb" to "obj/Debug/linksrc/ActivityTask.pdb".
    Copying file from "bin/Debug/HtmlParserSharp.pdb" to "obj/Debug/linksrc/HtmlParserSharp.pdb".
    Copying file from "bin/Debug/Mono.Android.pdb" to "obj/Debug/linksrc/Mono.Android.pdb".
Target _CopyMdbFiles:
  Skipping target "_CopyMdbFiles" because it has no outputs.
Target _StripEmbeddedLibraries:
  Skipping target "_StripEmbeddedLibraries" because all output files are up-to-date with respect to the input files.
Target _LinkAssembliesNoShrink:
    Touching "obj/Debug/link.flag".
Target _GenerateJavaStubs:
  Skipping target "_GenerateJavaStubs" because all output files are up-to-date with respect to the input files.
Target _AddStaticResources:
  Skipping target "_AddStaticResources" because all output files are up-to-date with respect to the input files.
Target _GeneratePackageManagerJava:
  Skipping target "_GeneratePackageManagerJava" because all output files are up-to-date with respect to the input files.
Target _CreateAdditionalResourceCache:
  Skipping target "_CreateAdditionalResourceCache" because it has no outputs.
Target _GenerateAndroidAssetsDir:
  Skipping target "_GenerateAndroidAssetsDir" because it has no outputs.
Target _CreateBaseApk:
  Skipping target "_CreateBaseApk" because all output files are up-to-date with respect to the input files.
Target _CreateAdditionalResourceCache:
  Skipping target "_CreateAdditionalResourceCache" because it has no outputs.
Target _CompileJava:
  Skipping target "_CompileJava" because all output files are up-to-date with respect to the input files.
Target _CreateAdditionalResourceCache:
  Skipping target "_CreateAdditionalResourceCache" because it has no outputs.
Target _CompileToDalvikWithDx:
    /usr/bin/java -Xmx1G -jar /Users/jeremie/android-sdk-macosx/build-tools/27.0.3/lib/dx.jar --dex --no-strict --output obj/Debug/android/bin /Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/android/bin/classes.zip /Library/Frameworks/Mono.framework/External/xbuild-frameworks/MonoAndroid/v8.1/mono.android.jar obj/Debug/lp/1/jl/bin/classes.jar obj/Debug/lp/11/jl/bin/classes.jar obj/Debug/lp/12/jl/bin/classes.jar obj/Debug/lp/13/jl/bin/classes.jar obj/Debug/lp/14/jl/bin/classes.jar obj/Debug/lp/15/jl/bin/classes.jar obj/Debug/lp/16/jl/bin/classes.jar obj/Debug/lp/17/jl/bin/classes.jar obj/Debug/lp/18/jl/bin/classes.jar obj/Debug/lp/19/jl/bin/classes.jar obj/Debug/lp/20/jl/bin/classes.jar obj/Debug/lp/21/jl/bin/classes.jar obj/Debug/lp/22/jl/classes.jar obj/Debug/lp/23/jl/classes.jar obj/Debug/lp/24/jl/classes.jar obj/Debug/lp/25/jl/classes.jar obj/Debug/lp/26/jl/classes.jar obj/Debug/lp/27/jl/classes.jar obj/Debug/lp/28/jl/classes.jar obj/Debug/lp/6/jl/arch-core-common.jar obj/Debug/lp/7/jl/arch-lifecycle-common.jar obj/Debug/lp/8/jl/bin/classes.jar obj/Debug/lp/9/jl/bin/classes.jar 
    Touching "obj/Debug/_dex_stamp".
Target _CopyPackage:
    Deleting file "bin/Debug/org.neteril.Moyeu.apk".
    Copying file from "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/android/bin/org.neteril.Moyeu.apk" to "bin/Debug/org.neteril.Moyeu.apk".
Target _ResolveAndroidSigningKey:
    Touching "obj/Debug/android_debug_keystore.flag".
Target _Sign:
    /usr/bin/keytool -list -alias androiddebugkey -storepass android -keypass android -keystore "/Users/jeremie/.local/share/Xamarin/Mono for Android/debug.keystore" 
    
    KEYTOOL : warning : 
    The JKS keystore uses a proprietary format. It is recommended to migrate to PKCS12 which is an industry standard format using "keytool -importkeystore -srckeystore /Users/jeremie/.local/share/Xamarin/Mono for Android/debug.keystore -destkeystore /Users/jeremie/.local/share/Xamarin/Mono for Android/debug.keystore -deststoretype pkcs12".
    Deleting file "bin/Debug/org.neteril.Moyeu-Signed.apk".
    /Users/jeremie/android-sdk-macosx/build-tools/27.0.3/zipalign 4 "/Users/jeremie/projects/Moyeu/Moyeu/obj/Debug/android/bin/org.neteril.Moyeu.apk" "bin/Debug//org.neteril.Moyeu-Signed.apk" 
    /usr/bin/java -jar /Users/jeremie/android-sdk-macosx/build-tools/27.0.3/lib/apksigner.jar sign --ks "/Users/jeremie/.local/share/Xamarin/Mono for Android/debug.keystore" --ks-pass pass:android --ks-key-alias androiddebugkey --key-pass pass:android --min-sdk-version 19 --max-sdk-version 27  /Users/jeremie/projects/Moyeu/Moyeu/bin/Debug/org.neteril.Moyeu-Signed.apk 
    Signed android package 'bin/Debug/org.neteril.Moyeu-Signed.apk'
Done building target "_Sign" in project "Moyeu.csproj".
Target _Upload:
    Detecting installed packages...

Detecting installed packages
    Removing old runtime: Mono.Android.DebugRuntime...


Removing old runtime
    Target device is arm64-v8a.
    Installing the Mono shared runtime (debug - 1520612195)...


Installing shared runtime
    10% ... 3520kb of 34672kb copied
    20% ... 6976kb of 34672kb copied
    30% ... 10432kb of 34672kb copied
    40% ... 13888kb of 34672kb copied
    50% ... 17344kb of 34672kb copied
    60% ... 20864kb of 34672kb copied
    70% ... 24320kb of 34672kb copied
    80% ... 27776kb of 34672kb copied
    90% ... 31232kb of 34672kb copied
    100% ... 34672kb of 34672kb copied
    Removing previous version of application...


Removing previous version of application
    Copying application to device...
    20% ... 960kb of 4610kb copied
    30% ... 1408kb of 4610kb copied
    40% ... 1856kb of 4610kb copied


Installing application on device
    70% ... 3264kb of 4610kb copied
    80% ... 3712kb of 4610kb copied
    90% ... 4160kb of 4610kb copied
    100% ... 4610kb of 4610kb copied
    Getting installation path...
    Using fast dev path: /storage/emulated/0/Android/data/org.neteril.Moyeu/files/.__override__
    Synchronizing assemblies...


Synchronizing assemblies
    0% ... 64kb of 12692kb copied
    10% ... 1283kb of 12692kb copied
    20% ... 2563kb of 12692kb copied
    30% ... 3834kb of 12692kb copied
    40% ... 5083kb of 12692kb copied
    50% ... 6402kb of 12692kb copied
    60% ... 7642kb of 12692kb copied
    70% ... 8923kb of 12692kb copied
    80% ... 10183kb of 12692kb copied
    90% ... 11455kb of 12692kb copied
    100% ... 12692kb of 12692kb copied
    Synchronizing native libs...
    Synchronizing dexes...


Synchronizing native libs


Synchronizing dexes
    Synchronizing resources...
    Synchronizing resources...
    Touching "obj/Debug/upload.flag".


Synchronizing resources


Synchronizing resources

Done building project "Moyeu.csproj".

Build succeeded.

KEYTOOL : warning : 
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:34.50
```

**Comment by shihaonan369** [comment]
how to fix it?I got same error.

**Comment by mrjavicho** [comment]
going to Settings-> All Apps  and uninstalling all Xamarin.Android API-* Support worked for me.

**Comment by balazsbonis** [comment]
I had to do what @mrjavicho did + set the linking behaviour to SDK assemblies only, and then it made my app deploy and work.

**Comment by tipa** [comment]
Having the same problem, the workaround provided by @mrjavicho doesnt work for me unfortunately

**Comment by andrekiba** [comment]
I have the same problem with my app released on Android 7.1 (I already set up SDK assemblies only)...

**Comment by jpobst** [comment]
The shared runtime was removed in Xamarin.Android 11.2, so I think we can close this.
https://github.com/xamarin/xamarin-android/releases#rewritten-fast-deployment-system-11199168
